File: cpp/grass.cpp
// grass.cpp
#include "metaclass.hpp"
#include <iostream>

class Grass : public Metaclass {
public:
    void displayAttributes() const override {
        std::cout << "Grass Attributes:\n"
                  << "Height: 10cm\n"
                  << "Color: Green\n"
                  << "Density: Medium\n"
                  << "Growth Condition: Humid\n"
                  << "Interaction: Can be trampled or cut\n";
    }
};

File: typescript/grass.ts
import { Metaclass } from "./metaclass";

export class Grass implements Metaclass {
  readonly type = "grass";

  describe() {
    return {
      height: "10cm",
      color: "green",
      density: "medium",
      growthCondition: "humid",
      interaction: ["trample", "cut"]
    };
  }
}
